README.md

```diff
@@ -0,0 +1,172 @@
+
+    // ... <code containing tracing events only sent to stdout>
+    // ...
+    // ...
+
+    // Time to enable telemetry from now on by setting telemetry=true on a `Span`
+    let enabled_span = span!(Level::INFO, "root", telemetry=true);
+    let _span_guard = enabled_span.enter();
+
+    info!("This event will be sent to InfluxDB");
+}
+```
+
+### Within Libraries
+
+**Warning: libraries should not set the global default subscriber as it will
+clobber any set by the application.**
+
+Libraries should not need to interact with `TelemetryLayer` directly. Instead,
+use the regular `tracing` crate macros (e.g `info!()`, `warn!()` etc) to record
+`Event`s.
+
+**Note:** `telemetry` is `disabled` by default. See below on how to enable telemetry
+on `Event`s within your library.
+
+### Detailed Examples
+
+For detailed examples, see `examples/*.rs`.
+
+## Enabling Telemetry
+
+Each `tracing` `Event` is either `enabled` or `disabled` based on the current
+`Span`'s setting (if one exists). By default, the root `Span` is disabled.
+
+There are 2 ways to set the `telemetry` field:
+
+1. On a `Span` itself e.g `span!(Level::ERROR, "span_name", telemetry=true)`
+1. Via function attributes e.g `#[tracing::instrument(fields(telemetry=true))]`
+
+Given that `tracing` `Span`s are hierarchical, if a function does not have a
+`tracing` attribute, or it does have one but the `telemetry` field is not set, we
+fall back by climbing the hierarchy until we explicitly find one set. In the end
+if no `Span` is found, we default to `telemetry=false`.
+
+## Warning!
+
+Be careful when using function attributes to enable telemetry as every function
+call invocation (along with its parameter names and values) will be a recorded
+as separate `Event`s and subsequently sent to InfluxDB...
+
+* If used within a hot path, lots of data will be generated and sent over the
+wire. Times this by the amount of installs across the entire userbase, and
+you'll see that it could be an overwhelming amount of data and it's eventual
+cost.
+
+* PII (Personally Identifiable Information) and other sensitive information
+  (think private keys and passwords) could make its way to InfluxDB by accident.
+  If your functions have any PII (including Public Keys or even hashes of unique
+  data etc), use the `skip()` function attribute i.e:
+
+  `#[tracing::instrument(skip(secret_key, password))]`
+
+## Debugging Telemetry
+
+### Inspecting Telemetry Files
+
+Telemetry files on disk are stored Base64 encoded. To peek into them:
+
+    cat ~/.fuelup/tmp/*.telemetry.* | while read line; do echo "$line" | base64 -d; echo; done
+
+### Manually Aging Out Telemetry Files
+
+As `FileWatcher` only submits files to InfluxDB that are over an hour old by
+default, we can force files to age out early so they are instantly submitted
+on next run:
+
+    for f in ~/.fuelup/tmp/*.telemetry.*; do mv "$f" "$f.old"; done
+    touch -t 202501010101 ~/.fuelup/tmp/*
```

Comment:
nit: also wrap in markdown `sh` here too

Reply:
Done.